internal/manifest/v1beta1/client.go

```diff
@@ -54,3 +61,36 @@ func (cc *ClusterClient) GetRESTConfig(
 	}
 	return restConfig, err
 }
+
+func WithClientCacheKey() declarative.WithClientCacheKeyOption {
+	cacheKey := func(ctx context.Context, resource declarative.Object) (any, bool) {
+		logger := log.FromContext(ctx)
+
+		if resource == nil {
```

Comment:
why is this case necessary?

Reply:
not necessary after a second look, just take from the previous code 😄